pkgs/applications/misc/blender/default.nix

```diff
@@ -1,243 +1,51 @@
-    ]
-    ++ lib.optionals waylandSupport [
-      "-DWITH_GHOST_WAYLAND=ON"
-      "-DWITH_GHOST_WAYLAND_DBUS=ON"
-      "-DWITH_GHOST_WAYLAND_DYNLOAD=OFF"
-      "-DWITH_GHOST_WAYLAND_LIBDECOR=ON"
-    ]
-    ++ lib.optionals stdenv.hostPlatform.isAarch64 [
-      "-DWITH_CYCLES_EMBREE=OFF"
-    ]
-    ++ lib.optionals stdenv.isDarwin [
-      "-DWITH_CYCLES_OSL=OFF" # requires LLVM
-      "-DWITH_OPENVDB=OFF" # OpenVDB currently doesn't build on darwin
+  latestStable = lib.last (builtins.sort lib.versionOlder (builtins.attrNames versions));
+  latestLTS = lib.last (builtins.sort lib.versionOlder (builtins.attrNames (lib.filterAttrs (_: v: v.isLTS == true) versions)));
 
-      "-DLIBDIR=/does-not-exist"
-    ]
-    # Clang doesn't support "-export-dynamic"
-    ++ lib.optional stdenv.cc.isClang "-DPYTHON_LINKFLAGS="
-    ++ lib.optional jackaudioSupport "-DWITH_JACK=ON"
-    ++ lib.optionals cudaSupport [
-      "-DWITH_CYCLES_CUDA_BINARIES=ON"
-      "-DWITH_CYCLES_DEVICE_OPTIX=ON"
-      "-DOPTIX_ROOT_DIR=${optix}"
-    ];
+  buildEnv = callPackage ./wrapper.nix {};
 
-  env.NIX_CFLAGS_COMPILE = "-I${ilmbase.dev}/include/OpenEXR -I${python}/include/${python.libPrefix}";
+  packages =
+    lib.mapAttrs'
+    (version: content: rec {
+      name = "blender_${version}";
+      value = callPackage ./derivation.nix {
+        inherit (darwin.apple_sdk.frameworks) Cocoa CoreGraphics ForceFeedback OpenAL OpenGL;
 
-  # Since some dependencies are built with gcc 6, we need gcc 6's
-  # libstdc++ in our RPATH. Sigh.
-  NIX_LDFLAGS = lib.optionalString cudaSupport "-rpath ${stdenv.cc.cc.lib}/lib";
+        # LLVM 11 crashes when compiling GHOST_SystemCocoa.mm
+        stdenv = if stdenv.isDarwin then llvmPackages_10.stdenv else stdenv;
 
-  blenderExecutable =
-    placeholder "out" + (if stdenv.isDarwin then "/Applications/Blender.app/Contents/MacOS/Blender" else "/bin/blender");
-  postInstall = lib.optionalString stdenv.isDarwin ''
-    mkdir $out/Applications
-    mv $out/Blender.app $out/Applications
-  '' + ''
-    mv $out/share/blender/${lib.versions.majorMinor version}/python{,-ext}
-    buildPythonPath "$pythonPath"
-    wrapProgram $blenderExecutable \
-      --prefix PATH : $program_PATH \
-      --prefix PYTHONPATH : "$program_PYTHONPATH" \
-      --add-flags '--python-use-system-env'
-  '';
+        # TODO Consider following https://vfxplatform.com/ for dependency tracking
+        # Like "VFX_RefCY": "2024", which outlines python 3.11, qt 6.5+ etc
+        python = python310;
 
-  # Set RUNPATH so that libcuda and libnvrtc in /run/opengl-driver(-32)/lib can be
-  # found. See the explanation in libglvnd.
-  postFixup = lib.optionalString cudaSupport ''
-    for program in $out/bin/blender $out/bin/.blender-wrapped; do
-      isELF "$program" || continue
-      addOpenGLRunpath "$program"
-    done
-  '';
+        withPackages = f: let packages = f value.python.pkgs; in buildEnv.override { blender = value; extraModules = packages; };
```

Comment:
I suspect that if one defines

```nix
anotherBlender = oldBlender.override { ... };
wrapped = anotherBlender.withPackages (ps: [ ... ]);
```

...then `wrapper.nix` might actually be called with `blender = oldBlender`, instead of `blender = anotherBlender`

Reply:
I'm confused, you mean that is wouldn't respect the override?

Reply:
Yes, you pass the `value`, but `anotherBlender` is `value.override { ... }` (I think)

Reply:
You could maybe just move this definition to `derivation.nix` and use `finalAttrs` instead of `blender`

Reply:
But also maybe I was wrong, so check first

Reply:
@SomeoneSerge I don't think you can even override python

Reply:
you're right, the override isn't used. I'll try to tfix